### PR TITLE
Add config agent core ngt service kvs test

### PR DESCRIPTION
**Pull request by hlts2**
Signed-off-by: hlts2 hiroto.funakoshi.hiroto@gmail.com

### Description:

I added test for `pkg/agent/core/ngt/service/kvs/test`.
and I added comment

**gramer check passed**

### Environment:

- Go Version: 1.16
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.12.3

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [x] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [ ] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [ ] I have added tests and benchmarks to cover my changes.
- [x] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: pkg/agent/core/ngt/service/kvs/kvs.go [hunk 1]
```diff
@@ -64,20 +66,27 @@ func New() BidiMap {
 	return b
 }
 
+// Get returns the value from the given key.
+// If the value does not exist, returns false.
```
Comment (hlts2) [inline pkg/agent/core/ngt/service/kvs/kvs.go:70]:
```suggestion
// Get returns the value from the given key.
// If the value does not exist, It returns false.
```

File: pkg/agent/core/ngt/service/kvs/kvs.go [hunk 2]
```diff
@@ -64,20 +66,27 @@ func New() BidiMap {
...
 func (b *bidi) Get(key string) (uint32, bool) {
 	return b.uo[xxhash.Sum64(stringToBytes(key))&mask].Load(key)
 }
 
+// Get returns the key from the given val.
+// If the key does not exist, returns false.
```
Comment (hlts2) [inline pkg/agent/core/ngt/service/kvs/kvs.go:76]:
```suggestion
// Get returns the key from the given val.
// If the key does not exist, It returns false.
```

File: pkg/agent/core/ngt/service/kvs/kvs.go [hunk 3]
```diff
@@ -64,20 +66,27 @@ func New() BidiMap {
...
 
+// Set sets the key and val.
 func (b *bidi) Set(key string, val uint32) {
 	b.uo[xxhash.Sum64(stringToBytes(key))&mask].Store(key, val)
 	b.ou[val&mask].Store(val, key)
 	atomic.AddUint64(&b.l, 1)
 }
 
+// Delete the value for the key and the key for the value.
+// If the value for the key does not exist, returns false.
```
Comment (hlts2) [inline pkg/agent/core/ngt/service/kvs/kvs.go:89]:
```suggestion
// Delete the value for the key and the key for the value.
// If the value for the key does not exist, It returns false.
```

File: pkg/agent/core/ngt/service/kvs/kvs.go [hunk 4]
```diff
@@ -90,6 +99,8 @@ func (b *bidi) Delete(key string) (val uint32, ok bool) {
 	return val, true
 }
 
+// Delete the key for the value and the value for the key a.
+// If the key for the val does not exist, returns false.
```
Comment (hlts2) [inline pkg/agent/core/ngt/service/kvs/kvs.go:103]:
```suggestion
// Delete the key for the value and the value for the key a.
// If the key for the val does not exist, It returns false.
```

File: pkg/agent/core/ngt/service/kvs/kvs.go [hunk 5]
```diff
@@ -64,20 +66,27 @@ func New() BidiMap {
 	return b
 }
 
+// Get returns the value from the given key.
+// If the value does not exist, It returns false.
```
Comment (kevindiu) [inline pkg/agent/core/ngt/service/kvs/kvs.go:70]:
```suggestion
// If the value does not exist, return false.
```

File: pkg/agent/core/ngt/service/kvs/kvs.go [hunk 6]
```diff
@@ -64,20 +66,27 @@ func New() BidiMap {
...
 func (b *bidi) Get(key string) (uint32, bool) {
 	return b.uo[xxhash.Sum64(stringToBytes(key))&mask].Load(key)
 }
 
+// Get returns the key from the given val.
+// If the key does not exist, It returns false.
```
Comment (kevindiu) [inline pkg/agent/core/ngt/service/kvs/kvs.go:76]:
```suggestion
// If the key does not exist, return false.
```

File: pkg/agent/core/ngt/service/kvs/kvs.go [hunk 7]
```diff
@@ -64,20 +66,27 @@ func New() BidiMap {
...
 
+// Set sets the key and val.
 func (b *bidi) Set(key string, val uint32) {
 	b.uo[xxhash.Sum64(stringToBytes(key))&mask].Store(key, val)
 	b.ou[val&mask].Store(val, key)
 	atomic.AddUint64(&b.l, 1)
 }
 
+// Delete the value for the key and the key for the value.
+// If the value for the key does not exist, It returns false.
```
Comment (kevindiu) [inline pkg/agent/core/ngt/service/kvs/kvs.go:89]:
```suggestion
// If the value for the key does not exist, return false.
```

File: pkg/agent/core/ngt/service/kvs/kvs.go [hunk 8]
```diff
@@ -90,6 +99,8 @@ func (b *bidi) Delete(key string) (val uint32, ok bool) {
 	return val, true
 }
 
+// Delete the key for the value and the value for the key a.
+// If the key for the val does not exist, It returns false.
```
Comment (kevindiu) [inline pkg/agent/core/ngt/service/kvs/kvs.go:103]:
```suggestion
// If the key for the val does not exist, return false.
```

File: pkg/agent/core/ngt/service/kvs/kvs_test.go [hunk 9]
```diff
@@ -121,62 +123,113 @@ func Test_bidi_Get(t *testing.T) {
...
+				fields: fields,
+				beforeFunc: func(_ args, bm BidiMap) {
+					bm.Set(key, val)
+				},
+				want: want{
+					want:  0,
+					want1: false,
+				},
+			}
+		}(),
```
Comment (kevindiu) [inline pkg/agent/core/ngt/service/kvs/kvs_test.go:211]: can we set empty string + val and able to get it?

File: pkg/agent/core/ngt/service/kvs/kvs_test.go [hunk 10]
```diff
@@ -121,62 +123,113 @@ func Test_bidi_Get(t *testing.T) {
...
+				fields.uo[i] = new(uo)
+			}
+
+			var (
+				key        = "45637ec4-c85f-11ea-87d0"
+				val uint32 = 14438
+			)
+
+			return test{
+				name:   "return (0, false) when there is no value for the key and the key is default value",
```
Comment (kevindiu) [inline pkg/agent/core/ngt/service/kvs/kvs_test.go:200]:
```suggestion
				name:   "return (0, false) when there is no value for the key and the key is empty string",
```

File: pkg/agent/core/ngt/service/kvs/kvs_test.go [hunk 11]
```diff
@@ -219,62 +272,114 @@ func Test_bidi_GetInverse(t *testing.T) {
...
+				fields.uo[i] = new(uo)
+			}
+
+			var (
+				key        = "45637ec4-c85f-11ea-87d0"
+				val uint32 = 14438
+			)
+
+			return test{
+				name:   "return false when there is a no key for the value and the val is default value",
```
Comment (kevindiu) [inline pkg/agent/core/ngt/service/kvs/kvs_test.go:349]:
```suggestion
				name:   "return false when there is a no key for the value and the val is 0",
```

File: pkg/agent/core/ngt/service/kvs/kvs_test.go [hunk 12]
```diff
@@ -219,62 +272,114 @@ func Test_bidi_GetInverse(t *testing.T) {
...
+				fields: fields,
+				beforeFunc: func(_ args, bm BidiMap) {
+					bm.Set(key, val)
+				},
+				want: want{
+					want:  "",
+					want1: false,
+				},
+			}
+		}(),
```
Comment (kevindiu) [inline pkg/agent/core/ngt/service/kvs/kvs_test.go:360]: can you add a test case to set the value 0 and get it?

File: pkg/agent/core/ngt/service/kvs/kvs_test.go [hunk 13]
```diff
@@ -295,57 +400,120 @@ func Test_bidi_Set(t *testing.T) {
...
+				fields.uo[i] = new(uo)
+			}
+
+			var (
+				key        = "45637ec4-c85f-11ea-87d0"
+				val uint32 = 14438
+			)
+
+			return test{
+				name: "set success",
```
Comment (kevindiu) [inline pkg/agent/core/ngt/service/kvs/kvs_test.go:453]:
```suggestion
				name: "set success when the key is not empty string and val is not 0",
```

File: pkg/agent/core/ngt/service/kvs/kvs_test.go [hunk 14]
```diff
@@ -295,57 +400,120 @@ func Test_bidi_Set(t *testing.T) {
...
+				fields.ou[i] = new(ou)
+				fields.uo[i] = new(uo)
+			}
+
+			var (
+				key = "45637ec4-c85f-11ea-87d0"
+			)
+
+			return test{
+				name: "set success when the val is default value",
```
Comment (kevindiu) [inline pkg/agent/core/ngt/service/kvs/kvs_test.go:480]:
```suggestion
				name: "set success when the val is 0",
```

File: pkg/agent/core/ngt/service/kvs/kvs_test.go [hunk 15]
```diff
@@ -295,57 +400,120 @@ func Test_bidi_Set(t *testing.T) {
...
+				fields.ou[i] = new(ou)
+				fields.uo[i] = new(uo)
+			}
+
+			var (
+				val uint32 = 14438
+			)
+
+			return test{
+				name: "set success when the key is default value",
```
Comment (kevindiu) [inline pkg/agent/core/ngt/service/kvs/kvs_test.go:506]:
```suggestion
				name: "set success when the key is empty string",
```

File: pkg/agent/core/ngt/service/kvs/kvs_test.go [hunk 16]
```diff
@@ -295,57 +400,120 @@ func Test_bidi_Set(t *testing.T) {
...
+				args: args{
+					val: val,
+				},
+				fields: fields,
+				want: want{
+					val: val,
+					l:   1,
+				},
+			}
+		}(),
```
Comment (kevindiu) [inline pkg/agent/core/ngt/service/kvs/kvs_test.go:516]: can you add a test case that both key and val is 0/empty value?

File: pkg/agent/core/ngt/service/kvs/kvs_test.go [hunk 17]
```diff
@@ -219,62 +272,114 @@ func Test_bidi_GetInverse(t *testing.T) {
...
+				fields.uo[i] = new(uo)
+			}
+
+			var (
+				key        = "45637ec4-c85f-11ea-87d0"
+				val uint32 = 14438
+			)
+
+			return test{
+				name: "return (45637ec4-c85f-11ea-87d0, true) when there is a key for the value",
```
Comment (kevindiu) [inline pkg/agent/core/ngt/service/kvs/kvs_test.go:290]:
```suggestion
				name: "return key and true when there is a key for the value",
```

File: pkg/agent/core/ngt/service/kvs/kvs_test.go [hunk 18]
```diff
@@ -121,62 +123,113 @@ func Test_bidi_Get(t *testing.T) {
...
+				fields.uo[i] = new(uo)
+			}
+
+			var (
+				key        = "45637ec4-c85f-11ea-87d0"
+				val uint32 = 14438
+			)
+
+			return test{
+				name: "return (14438, true) when there is a value for the key",
```
Comment (kevindiu) [inline pkg/agent/core/ngt/service/kvs/kvs_test.go:141]:
```suggestion
				name: "return the value when there is a value for the key",
```

File: pkg/agent/core/ngt/service/kvs/kvs_test.go [hunk 19]
```diff
@@ -409,65 +577,147 @@ func Test_bidi_Delete(t *testing.T) {
...
+				fields.uo[i] = new(uo)
+			}
+
+			var (
+				val uint32 = 14438
+			)
+			var wantl = 0
+
+			return test{
+				name:   "return (14438, true) when the delete successes and the key is default value",
```
Comment (kevindiu) [inline pkg/agent/core/ngt/service/kvs/kvs_test.go:640]:
```suggestion
				name:   "return val and true when the delete successes and the key is empty string",
```

File: pkg/agent/core/ngt/service/kvs/kvs_test.go [hunk 20]
```diff
@@ -409,65 +577,147 @@ func Test_bidi_Delete(t *testing.T) {
...
+			}
+
+			var (
+				key        = "45637ec4-c85f-11ea-87d0"
+				val uint32 = 14438
+			)
+			var wantl = 0
+
+			return test{
+				name: "return (14438, true) when the delete successes",
```
Comment (kevindiu) [inline pkg/agent/core/ngt/service/kvs/kvs_test.go:596]:
```suggestion
				name: "return val and true when the delete successes",
```

File: pkg/agent/core/ngt/service/kvs/kvs_test.go [hunk 21]
```diff
@@ -507,65 +757,147 @@ func Test_bidi_DeleteInverse(t *testing.T) {
...
+			}
+
+			var (
+				key        = "45637ec4-c85f-11ea-87d0"
+				val uint32 = 14438
+			)
+			var wantl = 0
+
+			return test{
+				name: "return (45637ec4-c85f-11ea-87d0, true) when the delete successes",
```
Comment (kevindiu) [inline pkg/agent/core/ngt/service/kvs/kvs_test.go:776]:
```suggestion
				name: "return key and true when the delete successes",
```

File: pkg/agent/core/ngt/service/kvs/kvs_test.go [hunk 22]
```diff
@@ -507,65 +757,147 @@ func Test_bidi_DeleteInverse(t *testing.T) {
...
+				fields.uo[i] = new(uo)
+			}
+
+			var (
+				key = "45637ec4-c85f-11ea-87d0"
+			)
+			var wantl = 0
+
+			return test{
+				name: "return (45637ec4-c85f-11ea-87d0, true) when the delete successes and the val is default value",
```
Comment (kevindiu) [inline pkg/agent/core/ngt/service/kvs/kvs_test.go:820]: same

**Review by kevindiu — CHANGES_REQUESTED** [review]

File: pkg/agent/core/ngt/service/kvs/kvs_test.go (hunk 9 above)
Comment (hlts2) [reply pkg/agent/core/ngt/service/kvs/kvs_test.go:211]: thank you 
I fixed: https://github.com/vdaas/vald/pull/1223/commits/7a47cef758f589a7a089f47cda801a7bfe918fe0

File: pkg/agent/core/ngt/service/kvs/kvs_test.go (hunk 16 above)
Comment (hlts2) [reply pkg/agent/core/ngt/service/kvs/kvs_test.go:543]: thank you
I fixed: https://github.com/vdaas/vald/pull/1223/commits/99c5dfc0c64934a9267dd8cee64732514242de36

File: pkg/agent/core/ngt/service/kvs/kvs_test.go (hunk 12 above)
Comment (hlts2) [reply pkg/agent/core/ngt/service/kvs/kvs_test.go:387]: thank you
I fixed: https://github.com/vdaas/vald/pull/1223/commits/b0dedf2f3a5f87df2b61fe28d778daffe4753f93

File: pkg/agent/core/ngt/service/kvs/kvs_test.go (hunk 22 above)
Comment (hlts2) [reply pkg/agent/core/ngt/service/kvs/kvs_test.go]: thank you
I fixed: https://github.com/vdaas/vald/pull/1223/commits/06e4a0ddeaaf71d1c1958a078f3de4e392eb9fb4

File: pkg/agent/core/ngt/service/kvs/kvs_test.go [hunk 23]
```diff
@@ -121,62 +123,140 @@ func Test_bidi_Get(t *testing.T) {
...
+				fields.uo[i] = new(uo)
+			}
+
+			var (
+				key        = ""
+				val uint32 = 0
+			)
+
+			return test{
+				name:   "return (0, false) when default value is set and there is value for the key and the key is default value",
```
Comment (kevindiu) [inline pkg/agent/core/ngt/service/kvs/kvs_test.go:227]:
```suggestion
				name:   "return (0, true) when the default value is set for the key and the key is empty string",
```

File: pkg/agent/core/ngt/service/kvs/kvs_test.go [hunk 24]
```diff
@@ -219,62 +299,141 @@ func Test_bidi_GetInverse(t *testing.T) {
...
+				fields.uo[i] = new(uo)
+			}
+
+			var (
+				key        = ""
+				val uint32 = 0
+			)
+
+			return test{
+				name:   "return true when set the default value and there is a key for the value and the val is 0",
```
Comment (kevindiu) [inline pkg/agent/core/ngt/service/kvs/kvs_test.go:403]:
```suggestion
				name:   "return (0, true) when the default value is set for the key and the val is 0",
```

**Review by kevindiu — COMMENTED** [review]
commented minor things.

File: pkg/agent/core/ngt/service/kvs/kvs_test.go [hunk 25]
```diff
@@ -121,62 +123,140 @@ func Test_bidi_Get(t *testing.T) {
...
+				fields: fields,
+				beforeFunc: func(_ args, bm BidiMap) {
+					bm.Set(key, val)
+				},
+				want: want{
+					want:  0,
+					want1: true,
+				},
+			}
+		}(),
```
Comment (kevindiu) [inline pkg/agent/core/ngt/service/kvs/kvs_test.go:238]: sorry again but can you add new test cases when the same key is set twice and see if it will override the value that is set?

File: pkg/agent/core/ngt/service/kvs/kvs_test.go [hunk 26]
```diff
@@ -219,62 +299,141 @@ func Test_bidi_GetInverse(t *testing.T) {
...
+				args:   args{},
+				fields: fields,
+				beforeFunc: func(_ args, bm BidiMap) {
+					bm.Set(key, val)
+				},
+				want: want{
+					want:  "",
+					want1: true,
+				},
+			}
```
Comment (kevindiu) [inline pkg/agent/core/ngt/service/kvs/kvs_test.go:413]: sorry again but can you add new test cases when the same key is set twice and see if it will override the value that is set?

File: pkg/agent/core/ngt/service/kvs/kvs_test.go (hunk 25 above)
Comment (hlts2) [reply pkg/agent/core/ngt/service/kvs/kvs_test.go:238]: thank you 
I fixed: https://github.com/vdaas/vald/pull/1223/commits/057528c256e881565370bccf7074687264ecb9b4

File: pkg/agent/core/ngt/service/kvs/kvs.go [hunk 27]
```diff
@@ -64,20 +66,27 @@ func New() BidiMap {
 	return b
 }
 
+// Get returns the value from the given key.
+// If the value does not exist, return false.
 func (b *bidi) Get(key string) (uint32, bool) {
 	return b.uo[xxhash.Sum64(stringToBytes(key))&mask].Load(key)
 }
 
+// Get returns the key from the given val.
```
Comment (hlts2) [inline pkg/agent/core/ngt/service/kvs/kvs.go:75]:
```suggestion
// GetInverse returns the key from the given val.
```

File: pkg/agent/core/ngt/service/kvs/kvs.go [hunk 28]
```diff
@@ -90,6 +99,8 @@ func (b *bidi) Delete(key string) (val uint32, ok bool) {
 	return val, true
 }
 
+// Delete the key for the value and the value for the key a.
+// If the key for the val does not exist, return false.
```
Comment (hlts2) [inline pkg/agent/core/ngt/service/kvs/kvs.go:103]:
```suggestion
// DeleteInverse the key and the value from the given val.
// If the key for the val does not exist, return false.
```

File: pkg/agent/core/ngt/service/kvs/kvs.go [hunk 29]
```diff
@@ -64,20 +66,27 @@ func New() BidiMap {
...
 }
 
+// Set sets the key and val.
 func (b *bidi) Set(key string, val uint32) {
 	b.uo[xxhash.Sum64(stringToBytes(key))&mask].Store(key, val)
 	b.ou[val&mask].Store(val, key)
 	atomic.AddUint64(&b.l, 1)
 }
 
+// Delete the value for the key and the key for the value.
```
Comment (hlts2) [inline pkg/agent/core/ngt/service/kvs/kvs.go:88]:
```suggestion
// Delete the key and the value from the given key.
```

**Review by kevindiu — APPROVED** [review]
LGTM

File: pkg/agent/core/ngt/service/kvs/kvs.go [hunk 30]
```diff
@@ -90,6 +99,8 @@ func (b *bidi) Delete(key string) (val uint32, ok bool) {
 	return val, true
 }
 
+// DeleteInverse the key and the value from the given val.
+// If the key for the val does not exist, return false.
```
Comment (vankichi) [inline pkg/agent/core/ngt/service/kvs/kvs.go:103]:
```suggestion
// DeleteInverse deletes the key and the value from the given val and returns the key and true.
// If the key for the val does not exist, it returns nil and false.
```

File: pkg/agent/core/ngt/service/kvs/kvs.go [hunk 31]
```diff
@@ -122,6 +134,7 @@ func (b *bidi) Range(ctx context.Context, f func(string, uint32) bool) {
 	wg.Wait()
 }
 
+// Len returns the length of the cache that is set.
```
Comment (vankichi) [inline pkg/agent/core/ngt/service/kvs/kvs.go:137]:
```suggestion
// Len returns the length of the cache that is set in the bidi.
```

**Review by vankichi — COMMENTED** [review]
You have to check the comment again

File: pkg/agent/core/ngt/service/kvs/kvs.go [hunk 32]
```diff
@@ -51,6 +52,7 @@ const (
 	// mask = 0xFFF.
 )
 
+// New returns BidiMap.
```
Comment (vankichi) [inline pkg/agent/core/ngt/service/kvs/kvs.go:55]:
```suggestion
// New returns the bidi that satisfies the BidiMap interface.
```

File: pkg/agent/core/ngt/service/kvs/kvs_test.go [hunk 33]
```diff
@@ -44,25 +45,26 @@ func TestNew(t *testing.T) {
...
+			var (
+				wantOu [slen]*ou
+				wantUo [slen]*uo
+			)
+			for i := 0; i < slen; i++ {
+				wantOu[i] = new(ou)
+				wantUo[i] = new(uo)
+			}
+			return test{
+				name: "return BindiMap",
```
Comment (vankichi) [inline pkg/agent/core/ngt/service/kvs/kvs_test.go:58]:
```suggestion
				name: "return the bidi struct",
```

File: pkg/agent/core/ngt/service/kvs/kvs_test.go [hunk 34]
```diff
@@ -121,62 +123,140 @@ func Test_bidi_Get(t *testing.T) {
...
+			for i := 0; i < slen; i++ {
+				fields.ou[i] = new(ou)
+				fields.uo[i] = new(uo)
+			}
+
+			var (
+				key1        = "45637ec4-c85f-11ea-87d0"
+				val1 uint32 = 14438
+			)
+			var key2 = "84a333-59633fd4-4553-414a"
```
Comment (vankichi) [inline pkg/agent/core/ngt/service/kvs/kvs_test.go:168]: could you use a more meaningful name?

File: pkg/agent/core/ngt/service/kvs/kvs_test.go (hunk 25 above)
Comment (vankichi) [reply pkg/agent/core/ngt/service/kvs/kvs_test.go:238]: Why don't you test when `l != 0`, `l = math.MaxUint64()`?

File: pkg/agent/core/ngt/service/kvs/kvs_test.go [hunk 35]
```diff
@@ -219,62 +299,141 @@ func Test_bidi_GetInverse(t *testing.T) {
...
+				fields: fields,
+				beforeFunc: func(_ args, bm BidiMap) {
+					bm.Set(key, val)
+				},
+				want: want{
+					want:  "",
+					want1: true,
+				},
+			}
+		}(),
```
Comment (vankichi) [inline pkg/agent/core/ngt/service/kvs/kvs_test.go:414]: same as the above comment

File: pkg/agent/core/ngt/service/kvs/kvs_test.go [hunk 36]
```diff
@@ -295,81 +454,199 @@ func Test_bidi_Set(t *testing.T) {
...
+			return test{
+				name:   "set success when the key and empty and the val is 0",
+				args:   args{},
+				fields: fields,
+				want: want{
+					val: 0,
+					l:   1,
+				},
+			}
+		}(),
```
Comment (vankichi) [inline pkg/agent/core/ngt/service/kvs/kvs_test.go:624]: same as the above test

File: pkg/agent/core/ngt/service/kvs/kvs_test.go [hunk 37]
```diff
@@ -409,65 +686,147 @@ func Test_bidi_Delete(t *testing.T) {
...
+				fields: fields,
+				beforeFunc: func(a args, bm BidiMap) {
+					bm.Set(key1, val1)
+				},
+				want: want{
+					wantVal: 0,
+					wantOk:  false,
+				},
+			}
+		}(),
```
Comment (vankichi) [inline pkg/agent/core/ngt/service/kvs/kvs_test.go:805]: same as the above comment

File: pkg/agent/core/ngt/service/kvs/kvs_test.go [hunk 38]
```diff
@@ -689,35 +1140,22 @@ func Test_bidi_Len(t *testing.T) {
...
-		*/
+		{
+			name: "return 100 when l of field is 100",
+			fields: fields{
+				l: 100,
+			},
+			want: want{
+				want: 100,
+			},
+		},
```
Comment (vankichi) [inline pkg/agent/core/ngt/service/kvs/kvs_test.go:1151]: same as the above

File: pkg/agent/core/ngt/service/kvs/kvs_test.go [hunk 39]
```diff
@@ -507,65 +866,147 @@ func Test_bidi_DeleteInverse(t *testing.T) {
...
+				beforeFunc: func(a args, bm BidiMap) {
+					bm.Set(key1, val1)
+				},
+				fields: fields,
+				want: want{
+					wantKey: "",
+					wantOk:  false,
+				},
+			}
+		}(),
```
Comment (vankichi) [inline pkg/agent/core/ngt/service/kvs/kvs_test.go:985]: same as the above

File: pkg/agent/core/ngt/service/kvs/kvs_test.go [hunk 40]
```diff
@@ -583,78 +1024,88 @@ func Test_bidi_Range(t *testing.T) {
...
+						return errors.Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", got, w.want)
+					}
+					return nil
+				},
+				fields: fields,
+				want: want{
+					want: wantMap,
+				},
+			}
+		}(),
```
Comment (vankichi) [inline pkg/agent/core/ngt/service/kvs/kvs_test.go:1087]: same as the above

File: pkg/agent/core/ngt/service/kvs/kvs_test.go (hunk 34 above)
Comment (hlts2) [reply pkg/agent/core/ngt/service/kvs/kvs_test.go]: thank you.
I fixed: https://github.com/vdaas/vald/pull/1223/commits/e660f2987b3ce4f611749b9d444445cfd10acc11

I deleted the variables that are not needed in the first place. And I added a meaningful variable name.:pray:

**Comment by kevindiu** [comment]
please fix ci fail.

File: pkg/agent/core/ngt/service/kvs/kvs.go [hunk 41]
```diff
@@ -64,20 +66,27 @@ func New() BidiMap {
...
 	return b.uo[xxhash.Sum64(stringToBytes(key))&mask].Load(key)
 }
 
+// GetInverse returns the key and the boolean from the given val.
+// If the key does not exist, it returns nil and false.
 func (b *bidi) GetInverse(val uint32) (string, bool) {
 	return b.ou[val&mask].Load(val)
 }
 
+// Set sets the key and val to the BidMap.
```
Comment (vankichi) [inline pkg/agent/core/ngt/service/kvs/kvs.go:81]:
```suggestion
// Set sets the key and val to the bidi.
```
sorry, I missed

File: pkg/agent/core/ngt/service/kvs/kvs.go [hunk 42]
```diff
@@ -64,20 +66,27 @@ func New() BidiMap {
...
 func (b *bidi) Set(key string, val uint32) {
 	b.uo[xxhash.Sum64(stringToBytes(key))&mask].Store(key, val)
 	b.ou[val&mask].Store(val, key)
 	atomic.AddUint64(&b.l, 1)
 }
 
+// Delete deletes the key and the value from the given key and returns val and true.
```
Comment (vankichi) [inline pkg/agent/core/ngt/service/kvs/kvs.go:88]:
```suggestion
// Delete deletes the key and the value from the bidi by the given key and returns val and true.
```

File: pkg/agent/core/ngt/service/kvs/kvs.go [hunk 43]
```diff
@@ -90,6 +99,8 @@ func (b *bidi) Delete(key string) (val uint32, ok bool) {
 	return val, true
 }
 
+// DeleteInverse deletes the key and the value from the given val and returns the key and true.
```
Comment (vankichi) [inline pkg/agent/core/ngt/service/kvs/kvs.go:102]:
```suggestion
// DeleteInverse deletes the key and the value from the bidi by the given val and returns the key and true.
```

**Review by kevindiu — DISMISSED** [review]
LGTM

**Review by kevindiu — APPROVED** [review]

**Review by vankichi — APPROVED** [review]
